app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/MapDataWithEditsSource.kt

```diff
@@ -184,6 +186,15 @@ class MapDataWithEditsSource internal constructor(
                         // element that got edited by the deleted edit not found? Hmm, okay then (not sure if this can happen at all)
                         elementsToDelete.add(ElementKey(edit.elementType, edit.elementId))
                     }
+
+                    if (edit.action is MoveNodeAction || edit.action is RevertMoveNodeAction) {
+                        val waysContainingNode = getWaysForNode(edit.elementId)
+                        val affectedRelations = getRelationsForNode(edit.elementId) + waysContainingNode.flatMap { getRelationsForWay(it.id) }
+                        for (elem in waysContainingNode + affectedRelations) {
+                            mapData.put(elem, getGeometry(elem.type, elem.id))
+                        }
+                    }
+
```

Comment:
What is this for? I thought MoveNodeAction would only work for free-floating nodes?

Also, in any case, unit tests would be very useful here.

Reply:
It's used only for free floating nodes, but the action works also for nodes that are part of a way or relation.
And I think allowing some (non-multipolygon) relations might actually make sense, e.g. for moving a bus stop.

Reply:
Oh okay, that is quite foresighted of you. But then, please also add tests that confirm this behavior.